integration/tsconfig-test.ts

```diff
@@ -0,0 +1,125 @@
+import { test, expect } from "@playwright/test";
+import fse from "fs-extra";
+import path from "path";
+import JSON5 from "json5";
+
+import { createFixture, json } from "./helpers/create-fixture";
+
+async function getTsConfig(projectDir: string) {
+  let tsconfigPath = path.join(projectDir, "tsconfig.json");
+  let config = await fse.readFile(tsconfigPath, "utf8");
+  return JSON5.parse(config);
+}
+
+// this is the default tsconfig.json that is shipped with `create-remix` templates
+const DEFAULT_CONFIG = {
+  include: ["remix.env.d.ts", "**/*.ts", "**/*.tsx"],
+  compilerOptions: {
+    allowJs: true,
+    esModuleInterop: true,
+    forceConsistentCasingInFileNames: true,
+    isolatedModules: true,
+    jsx: "react-jsx",
+    lib: ["DOM", "DOM.Iterable", "ES2019"],
+    moduleResolution: "node",
+    noEmit: true,
+    resolveJsonModule: true,
+    strict: true,
+    target: "ES2019",
+    baseUrl: ".",
+    paths: {
+      "~/*": ["./app/*"],
+    },
+  },
+};
```

Comment:
after https://github.com/remix-run/remix/pull/3013 is merged we can do this instead 


```ts
const DEFAULT_CONFIG_CONTENT = fse.readFileSync(
  path.join(process.cwd(), "templates/remix/tsconfig.json"),
  "utf8"
);

const DEFAULT_CONFIG = JSON5.parse(DEFAULT_CONFIG_CONTENT);
```

Reply:
I'd rather not use stuff in `templates`. It's unclear to future maintainers that those things are used in tests and false breakages could lead to confusion. I'm fine with the object here.